src/cmd/flux-job.c

```diff
@@ -857,27 +857,38 @@ int cmd_wait_event (optparse_t *p, int argc, char **argv)
 }
 
 struct info_ctx {
-    optparse_t *p;
     flux_jobid_t id;
-    const char *key;
+    json_t *keys;
 };
 
-void info_continuation (flux_future_t *f, void *arg)
+void info_output (flux_future_t *f, const char *suffix, flux_jobid_t id)
 {
-    struct info_ctx *ctx = arg;
     const char *s;
 
-    if (flux_rpc_get_unpack (f, "{s:s}", ctx->key, &s) < 0) {
+    if (flux_rpc_get_unpack (f, "{s:s}", suffix, &s) < 0) {
         if (errno == ENOENT) {
             flux_future_destroy (f);
-            log_msg_exit ("job %lu.%s not found", ctx->id, ctx->key);
+            log_msg_exit ("job %lu id or key not found", id);
```

Comment:
Might be nice to return a text string from the service so that one knows whether the job or the key is wrong.

Reply:
part of the issue is that we don't know.  It's simply an ENOENT from the KVS.  It could be b/c the jobid is missing or the key is missing.

Hmmm.  Since the infrastructure does everything in parallel now, we could add an additional lookup on just `jobs.active.jobid` and that can give us info for an error return string?

Reply:
Perhaps if the eventlog is missing, we know it's a bad jobid since every valid jobid has an eventlog from the very start?  If eventlog is valid and key isn't then the key is likely missing.

Reply:
Hmmm. That would require us to lookup the eventlog even when we don't need to (i.e. we already checked that the instance owner sent the rpc).  

If we looked up the eventlog everytime, i guess it wouldn't be the end of the world.  Would we want to add it just for the nicer error msg?

Reply:
Ooops I forgot that we don't read it every time.  Well, for now maybe not worth it.

However it may still be useful to return a textual error message.  For example, if lookup of a particular key returns ENOENT we could at least say which one of the set caused the failure.

Reply:
Or maybe it is worth it.  I was thinking of the case where a queue listing tool needs to fetch info for every job, where the extra kvs lookups might add up to something substantial.  But I suppose in that case, most likely the tool is going to fetch everything anyway.  In the single job case, fetching it in parallel with other things won't be a big deal.

What is your opinion?

Reply:
that makes sense, lets fetch it everytime.

Reply:
This ended up trickier than I had hoped.  Because `active` to `inactive` transition can happen at anytime, I believe it's possible to get success on one lookup and ENOENT on another lookup done in parallel.

So what I've been doing prior to this is "switch" to inactive whenever I get an ENOENT on any key that is missing in the `active` path.

Assume a job is still active.  What if the user inputs `flux job info 12345 eventlog jobspec badkey`.  The `job-info` module notices ENOENT on `badkey`, switches to `inactive` and notices `eventlog` is missing in `inactive`.  Now I need to manage that I found `eventlog` in the active path prior.  Also I have to iterate through all keys to check if `eventlog` was found or not.

I think looking up a `jobs.active.jobid` would be the right path now.  We can also drop this issue until later.

Reply:
It seems like it's the same problem whether we look up `jobs.active.jobid` or `jobs.active.jobid.eventlog`.  Both would transition in the same way.

This inactive archival plan needs a rethink (as discussed last week with @grondo), so since we haven't implemented it yet anyway, maybe just remove the active/inactive logic from this part of the code now.

Sorry this design proved to be such a pain.

Reply:
The PR is getting a bit long, so I'm going to start splitting it up.  How about we save inactive removal for another PR.

Reply:
Fine by me.